src/configuration.rs

```diff
@@ -0,0 +1,376 @@
+
+
+use ed25519_dalek::*;
+//use rand::{Rng, RngCore, CryptoRng};
+use rand;
+use std::time::{Duration, SystemTime};
+use crypto::{
+    //aead::AeadEncryptor, chacha20poly1305::ChaCha20Poly1305,
+    aes, blockmodes,
+    buffer::{ ReadBuffer, WriteBuffer, BufferResult, RefReadBuffer, RefWriteBuffer },
+    sha2, digest::Digest,
+};
+
+use crate::error::*;
+
+// See: lib3h/crates/sodium/src/pwhash.rs, crypto_pwhash_argon2id_{OPS,MEM}LIMIT_INTERACTIVE
+pub const OPSLIMIT_INTERACTIVE: u32 = 2;
+pub const MEMLIMIT_INTERACTIVE: u32 = 1 << 16; // ~67MB
```

Comment:
I feel that this should be inlined. `OPSLIMIT_INTERACTIVE`, `MEMLIMIT_INTERACTIVE` and so on are libsodium convention names, and we don't use libsodium.